### DISCUSSION TITLE
Ability to create separate areas of collections

### DISCUSSION THREAD

**Discussion by Seinnd** [directus/directus#19627] [Feature Requests]
### Summary

Advanced grouping function for additional sidebar appearance with spaces/areas for grouped or single collections, or allow groups (folders) to act as a separated space/area to greatly improve usability and customization options.

### Motivation

Even in medium-sized projects, I have often noticed that a small visual update would greatly simplify the handling of data.

The explicit issue is that collections can only be sorted into groups (folders), but not as a separate area.
If collections and groups could be separated into areas, this would extremely increase the thematic distinctions and the handling of data in terms of usability and accessibility.

Moreover, taking data redundancy into account, it would be possible to separate technically different data structures in this way without having to give up systemic relations.
In keeping with the "single-source-of-trouth" idea, many different topics can be brought together in one system in this way without usability suffering - as is the case with a very long list (grouping or not) as a single data model.

In this way, it is also easier to visualise the topic of "multi-tenancy" or "multi-site" without having to invest a great deal of non-structural effort in logical groups (folders) for role distribution or even the synchronisation of different databases or Directus installations.

### Detailed Design

![01_create_space](https://github.com/directus/directus/assets/22862290/af8d91e2-1787-4983-a41c-21a89e6774fc)
possibility to create not only groups (folders) but also separate areas.

or

![02_use_as_section](https://github.com/directus/directus/assets/22862290/014c1242-62f1-4549-bff5-729327724b68)
e.g. use groups (folders) with additional appearance option

which lead to the following visual difference in the UI compared to a simple group (folder), where also only the assigned groups/collections are listed:
![03_usability_improvement](https://github.com/directus/directus/assets/22862290/9e2ed7de-a160-4041-950f-f847d15c1964)

### Must Have:
- visual delineation of subject areas of the data model

### Should Have:
- additional grouping option for collection folders

### Could Have:
- create complete new space for collections instead of grouping just by folders

### Won't Have:
- I dunno. Maybe the fact to just leave it as it is, because for non-technical users this kind of data presentation is rather cluttered and hard to manage as endless non-demarcated topics. 

### Drawbacks

I'm not that deeply involved in what's going on under the bonnet of Directus, but my basic idea integrates directly with the existing possibilities and should therefore have little to no drawbacks - from my point of view as a pure user, it's rather the opposite, as Directus stands out from all other tools for data visualisation (which also just output a pure endless list of a data model). 

### Alternatives

If the above approach means too much work of implementation, an even simpler solution would be to include both the name of a "custom link" in the "Module Bar" and the group (folder) names of collections as ID/class inside the DOM.

In this way, the desired effect can be created dynamically with a few lines of CSS.

### Adoption Strategy

Since the basic idea is a purely visual adaptation to make it easier to delineate data structures and relationships (without losing the possibility of relations and dependencies), I can imagine that this will be very well received by the community, since no direct changes would have to be made to collections per se.

**Comment by vanling** [comment]
Yes, would be awesome to have in Directus :D

**Reply by u12206050** [reply]
Would be nice to add custom modules to specific sections. Ie. When you hover or click on "Human Resources" you get a slide out menu with "Content" (for the default content module) and any other  modules you want for that section.

**Reply by Seinnd** [reply]
There is still more to be done to increase user-friendliness. 
The above request is only a first easily realisable idea for the visual separation of different subject areas. 
At the moment I could just use PHPMyAdmin to manage a data model, like in Directus, without any disadvantage (from the point of view of end users) - which is a pity, because Directus has so much potential to be more!

**Comment by rijkvanzanten** [comment]
Heya! Thanks for opening this feature request!

This feature request has received over **15 votes** from the community. This means we'll move this feature request to the Under Review state! 

The Core team will schedule a meeting to review this request as soon as possible. The discussion will then be approved or denied. You may or may not be invited to join this meeting with the core team.

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Comment by ghost** [comment]
With an extension we're building at my workplace we've encountered the need to manually duplicate several non-exported composables and components from the Directus core. This workaround has been necessary in order to recreate the layout, search, filter, and pagination functionalities found in the 'Content' interface within our custom menu area. 
![Screenshot 2023-09-25 at 4 49 06 PM](https://github.com/directus/directus/assets/113140474/8b286861-079f-4e6a-8ae4-6903cbd841d2)

Copying core files into our extension is an ugly way to manage the code. But this approach has been necessary due to hesitancy to export more core components and composables from the Directus core on account of the potential risk of introducing breaking changes for extensions, as discussed in issue #17234 .

This feature request seems like a great middle-ground where extensions would be able to use the built-in framework that makes Directus amazing to use while still maintaining some degree of separation between my site's "Content" and the collections which are explicitly related to configuring and managing our extensions.

Components & composables we're manually copying from Directus core, just to achieve the functionality that this request would provide include:
- search-input.vue
- tabular.vue
- archive-sidebar-detail.vue
- layout-sidebar-detail.vue
- information-sidebar-detail.vue
- use-page-size.ts
- use-preset.ts
- use-notify.ts
- use-alias-fields.ts

**Comment by Seinnd** [comment]
Since this request has now been moved to the "under review" category, I assume that the idea behind a topic-based orientation will find favour. If this idea is carried forward, then there should also be the possibility to assign insights as entry level, and also files and extensions directly to these areas. 

![image](https://github.com/directus/directus/assets/22862290/cc91da09-ba31-4f4e-920e-79ad3fc5b79d)

The approach of a topic-based structure, under which everything is assigned, would also considerably simplify rights management for large databases/-models (I suspect at least). 
So instead of controlling hundreds of collections individually (over and over again while the project keeps growing), only the respective area would/could then be controlled and possibly only in individual cases deeper at collection level (exclude instead of an include mechanism).
